test/products.service.spec.ts

```
@@ -0,0 +1,349 @@
+      sort: jest.fn().mockReturnThis(),
+      skip: jest.fn().mockReturnThis(),
+      limit: jest.fn().mockReturnThis(),
+      lean: jest.fn().mockReturnThis(),
+      exec: jest.fn(),
+      save: jest.fn(),
+      or: jest.fn().mockReturnThis(),
+      aggregate: jest.fn().mockReturnThis(),
+    };
+
+    // For "new productModel()" calls
+    mockProductModel.constructor = jest.fn().mockImplementation(() => ({
```

ESLint

[error] Immediate mutation on object is not allowed.

(unicorn/no-immediate-mutation)